cmd/argocd/commands/admin/project_allowlist.go

```diff
@@ -63,7 +63,13 @@ func NewProjectAllowListGenCommand() *cobra.Command {
 				}()
 			}
 
-			globalProj := generateProjectAllowList(clientConfig, clusterRoleFileName, projName)
+			config, err := clientConfig.ClientConfig()
+			errors.CheckError(err)
+			disco, err := discovery.NewDiscoveryClientForConfig(config)
+			errors.CheckError(err)
+			serverResources, err := disco.ServerPreferredResources()
+			errors.CheckError(err)
+			globalProj := generateProjectAllowList(serverResources, clusterRoleFileName, projName)
```

Comment:
I don't think the logic to retrieve the resourceList should be added directly in the `NewProjectAllowListGenCommand`. Ideally, commands should be very dry in terms of logic mainly dealing with CLI flags and delegating the execution to an underlying function like it was before. Another possibility to achieve the same goal is defining an interface like:
```go
type ResourceGetter interface {
    GetServerResources(c clientcmd.ClientConfig) ([]*metav1.APIResourceList, error)
}
```
And then have this interface as the first argument of `generateProjectAllowList()`. This way you can easily mock this logic and properly test it.